### ISSUE TITLE
Unable to copy database to clipboard in Chome

### ISSUE THREAD

**Issue by Pustur** [frzyc/genshin-optimizer#70]
When trying to save my DB to the clipboard in [this page](https://frzyc.github.io/genshin-optimizer/#/database), I press the button, the alert box comes up saying `Copied database to clipboard` but nothing is actually saved to the clipboard.

In the console the following error occurs:

```
Uncaught (in promise) DOMException: Document is not focused.
```

It works fine in Firefox and Safari, but not Chrome.

I don't understand the error, if I'm pressing the button -> the button is focused -> the document should be focused too no? So what's up?

**Comment by pha-Z** [comment]
using chrome and copying to clipboard works fine for me..

**Comment by Pustur** [comment]
So I tested it with my windows PC and it works there, but on my mac it doesn't.

Here's my specs:

- **Chrome**  
Version 93.0.4577.63 (Official Build) (x86_64)
- **macOS Catalina**  
10.15.7

@pha-Z Did you test on a windows or mac?

**Comment by pha-Z** [comment]
Windows

**Comment by Pustur** [comment]
Ok I think I understand the problem now.

In this code:

https://github.com/frzyc/genshin-optimizer/blob/ce5919de268b61179c811e0882b57951170f06a0/src/Settings/SettingsDisplay.tsx#L93-L94

You try to copy the DB to the clipboard and then alert right after it. The alert is actually the thing that's responsible for the error! It basically steals the focus from the document so the clipboard operation fails. (apparently this focus problem only occurs on Chrome + macOS for some reason)

Instead the correct code would be:

```js
navigator.clipboard.writeText(JSON.stringify(exportGOOD(dbStorage)))
  .then(() => alert("Copied database to clipboard."))
  .catch(console.error)
```

So the alert is only shown once the clipboard has been written to successfully.

**Comment by frzyc** [comment]
@Pustur Sorry for the late reply, I had to prioritize time on more critical features. What a surprise for me to find the solution already waiting for me. The problem is resolved as of `v6.0.13`. 

If you want more up-to-date responses to bugreports, I'll advice you to join the [Genshin Optimizer Discord](https://discord.gg/8Hpz2F7AnR), where I stage most of the feature requests/bugreports/discussions. 

Thanks a lot for your help!
